src/cli/commands/build.ts

```diff
@@ -0,0 +1,173 @@
+    }
+  });
+
+  let appLocation = options.appLocation !== undefined ? options.appLocation : workflowConfig?.appLocation;
+  let apiLocation = options.apiLocation !== undefined ? options.apiLocation : workflowConfig?.apiLocation;
+  let outputLocation = options.outputLocation !== undefined ? options.outputLocation : workflowConfig?.outputLocation;
+  let appBuildCommand = options.appBuildCommand !== undefined ? options.appBuildCommand : workflowConfig?.appBuildCommand;
+  let apiBuildCommand = options.apiBuildCommand !== undefined ? options.apiBuildCommand : workflowConfig?.apiBuildCommand;
+
+  if (options.auto && hasBuildOptionsDefined(options)) {
+    logger.error(`You can't use the --auto option when you have defined appBuildCommand or apiBuildCommand in ${swaCliConfigFilename}`);
+    logger.error(`or with the --app-build-command and --api-build-command options.`, true);
+    return;
+  }
+
+  if (options.auto) {
+    logger.log('Detecting build configuration...');
+    const detectedFolders = await detectProjectFolders(appLocation);
+
+    if (detectedFolders.app.length === 0 && detectedFolders.api.length === 0) {
+      logger.error(`Your app configuration could not be detected.`);
+      return showAutoErrorMessageAndExit();
+    } else if (detectedFolders.app.length > 1 || detectedFolders.api.length > 1) {
+      logger.error(`Multiple apps found in your project folder.`);
+      return showAutoErrorMessageAndExit();
+    }
+
+    let projectConfig;
+    try {
+      projectConfig = await generateConfiguration(detectedFolders.app[0], detectedFolders.api[0]);
+      appLocation = projectConfig.appLocation;
+      apiLocation = projectConfig.apiLocation;
+      outputLocation = projectConfig.outputLocation;
+      apiBuildCommand = projectConfig.apiBuildCommand;
+      appBuildCommand = projectConfig.appBuildCommand;
+    } catch (error) {
+      logger.error(`Cannot generate your build configuration:`);
+      logger.error(error as Error, true);
+      return;
+    }
+  }
+  
+  if (!appBuildCommand && !apiBuildCommand) {
+    if (!hasBuildOptionsDefined(options)) {
+      logger.warn('No build options were defined.');
+      logger.warn(`Please "swa init" to set your configuration or use options flags to set your`);
+      logger.warn(`build commands and paths.\n`);
+    }
+
+    logger.log('Nothing to build.');
+    return;
+  }
+
+  logger.log(`Build configuration:`);
+  logger.log(`- App location: ${chalk.green(appLocation || '')}`);
+  logger.log(`- API location: ${chalk.green(apiLocation || '')}`);
+  logger.log(`- Output location: ${chalk.green(outputLocation || '')}`);
+  logger.log(`- App build command: ${chalk.green(appBuildCommand || '')}`);
+  logger.log(`- API build command: ${chalk.green(apiBuildCommand || '')}`);
+  
+  if (appBuildCommand) {
+    let buildPath = appLocation!;
+    const packageJsonPath = await findUpPackageJsonDir(appLocation!, outputLocation!);
+    if (packageJsonPath) {
```

Comment:
what about python and dotnet?

Reply:
They will run from appLocation by default, but this is not ideal.
We should open a separate issue to address how we find the root app and api paths for all stacks, to use that with "swa build" and "swa start" (as it's also needed for the --run flag)